### DISCUSSION TITLE
How to create an empty session setup?

### DISCUSSION THREAD

**Discussion by ghost** [jamescherti/easysession.el#11] [Q&A] [answered]
How can I configure easysession to create a minimal environment when a new session is started? I want to ensure that all windows and tabs are closed, leaving only the scratch buffer open.

**Comment by jamescherti** [comment, marked as answer]
To set up a minimal environment when easysession creates a new session, you can define a function that closes all other tabs, deletes all other windows, and switches to the scratch buffer. The following Emacs Lisp code demonstrates how to achieve this:
``` emacs-lisp
(defun my-empty-easysession ()
  "Set up a minimal environment when easysession creates a new session."
  (when (and (boundp 'tab-bar-mode) tab-bar-mode)
    (tab-bar-close-other-tabs))
  (delete-other-windows)
  (scratch-buffer))

(add-hook 'easysession-new-session-hook #'my-empty-easysession)
```
